api/v1alpha1/dnsrecord_types_test.go

```diff
@@ -58,17 +53,6 @@ func TestValidate(t *testing.T) {
 			},
 			wantErr: false,
 		},
-		{
-			name:     "valid wildcard domain no endpoint",
-			rootHost: "*.example.com",
-			dnsNames: []string{
-				"a.b.example.com",
-				"b.a.example.com",
-				"a.example.com",
-				"b.example.com",
-			},
-			wantErr: true,
-		},
```

Comment:
This should error, it would expect an endpoint to have a "*.example.com" dnsName. I think you removed the check and added it back so this can probably be reverted?

Reply:
I had to keep that check out [here](https://github.com/Kuadrant/dns-operator/pull/504/files#diff-6a70ad61b66d79fb40a3db3e374825803cbd470e8f37a79153c054aace8b5527L249) as the zone record might be e.g. `rootHost: pb.hcpapps.net`, but the records merged into it: `app.pb.hcpapps.net`, so this error would trigger then and block the reconcile.

Reply:
as it stands, the endpoint provider uses the existing logic to find a matching zone for the record (i.e. looks for the closest matching zone name), so it's possible to match a DNS Record `app.pb.hcpapps.net` against a zone record of `pb.hcpapps.net`. Although it will also work if both are `app.pb.hcpapps.net`.

I suppose it depends on the plan for creating the authoritative record, if it'll always match, then I can change the "find the right zone" logic for the endpoints to always expect a direct match, and then reinstate the above check.

I'm happy to go either way, but I made these changes under the assumption that we wanted the existing zone finding logic used in AWS etc, to also apply to the endpoints provider.

Reply:
Spoke offline and now see what the issue is, and why we want to remove this validation.

Thought it through and i don't think it should be an issue. The important thing with this is that all the endpoints in the slice are related, and are part of the same domain as the `rootHost`, which we are still validating for.

Reply:
Discussed in person: as we want to allow source records to be a subdomain of the destination records  (e.g. source rootHost: app.pb.com and destination rootHost: pb.com), we will leave this validation step out.